jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -354,32 +354,30 @@ public void scanJars(final WebAppContext context, Collection<Resource> jars, boo
         //Scan jars for META-INF information
         if (jars != null)
         {
-            try (ResourceFactory.Closeable scanResourceFactory = ResourceFactory.closeable())
+            ResourceFactory resourceFactory = ResourceFactory.of(context);
```

Comment:
Use `context.getResourceFactory()` instead?
I would also add a comment explaining why this factory must be kept open after this method returned.

Reply:
This ResourceFactory is deliberately temporary because we do not want to pollute the mounted resources with ones that ultimately do not contain any web fragments, tlds etc etc.

Reply:
@janbartel is making a very good point which is worth a code comment IMHO.

But if the resources that are found survive the scope of this closeable resource factory, their jars should be mounted by the context' resource factory before closing the closeable one to guarantee they're always useable during the lifecycle of the context.

Reply:
@lorban the jars that contain stuff worthwhile mounting by the context's resource factory is handled by line https://github.com/jetty/jetty.project/blob/jetty-12.0.x/jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java#L425. Until we know if it's worthwhile mounting, we don't want to pollute the space with jars that we don't need to scan/don't contain any META-INF/tlds/fragments. I think your concerns have been addressed, but agree that a comment on this line would be useful so we don't forget why we've done this in the future.